src/sentry/services/hybrid_cloud/organization/impl.py

```diff
@@ -286,6 +289,7 @@ def add_organization_member(
             flags=self._deserialize_member_flags(flags) if flags else 0,
             role=role or organization.default_role,
         )
+        organizationmember_mapping_service.create_with_organization_member(member)
```

Comment:
What happens if the call to `create_with_organization_member` fails (network partition, database failure, validation failure on the row)?  Seems like you end up with an OrganizationMember that does not ever receive an organizationmember_mapping object.

Reply:
@corps would putting this block under `with transaction.atomic()` be sufficient?

Reply:
@dashed Ah hah!  Not quite -- this is an important distinction.  In tests, yes that will work **incidentally** only because the service will be in process currently, but after @RyanSkonnord 's work on RPC makes more progress, we may need to *intentionally break any active transaction whilst inside of a service call!*.

Here's the trick, we need to imagine that any service call is actually a remote network call.  Which, means, *local database transactions cannot span that remote call!*.  In fact, no matter how you re-arrange this, *it will always be possible that the network call fails*.  You may think that is fine, if the network call fails you can fail your local transaction, too... except.

**Even if a network call fails, that does not tell if you if the remote transaction itself failed, only the communication of its status**.  It's possible the other side DID commit the new org member mapping, so if you fail the current transaction, you still end up with integrity issues.

This is the core of the distributed systems problem; consistency is not free.

Fortunately, I think there is a fix and pattern here.  We need to create an outbox containing instructions to create the org member mapping inside the transaction we create org member itself.  Let me know when you want to pair on this, I can show you da way.